### PR TITLE
feat: add types for @npmcli/promise-spawn

### PR DESCRIPTION
**Pull request by antongolub**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by antongolub** [comment]
CC @lukekarrys, @nlf

File: types/npmcli__promise-spawn/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (peterblazejewicz) [inline types/npmcli__promise-spawn/package.json:1]: IMO this one is not required really, @types/node are installed by automatic types acquisition (in supporting IDEs) or explicitly by user.

File: types/npmcli__promise-spawn/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// Type definitions for @npmcli/promise-spawn 6.0
+// Project: https://github.com/npm/promise-spawn#readme
+// Definitions by: Anton Golub <https://github.com/antongolub>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.3
```
Comment (peterblazejewicz) [inline types/npmcli__promise-spawn/index.d.ts:5]:
```suggestion

```
why this one was added? I cannot see nothing specific to TS4.3 or newer

File: types/npmcli__promise-spawn/package.json (hunk 1 above)
Comment (antongolub) [reply types/npmcli__promise-spawn/package.json]: Fixed.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@antongolub thanks!

**Comment by antongolub** [comment]
Ready to merge

**Pull request merged**